pandas/core/indexes/interval.py

```diff
@@ -189,161 +127,63 @@ class IntervalIndex(IntervalMixin, Index):
 
     See further examples in the doc strings of ``interval_range`` and the
     mentioned constructor methods.
+    """),
 
-    Notes
-    ------
-    See the `user guide
-    <http://pandas.pydata.org/pandas-docs/stable/advanced.html#intervalindex>`_
-    for more.
-
-    See Also
-    --------
-    Index : The base pandas Index type
-    Interval : A bounded slice-like interval; the elements of an IntervalIndex
-    interval_range : Function to create a fixed frequency IntervalIndex
-    cut, qcut : Convert arrays of continuous data into Categoricals/Series of
-                Intervals
-    """
+))
+class IntervalIndex(IntervalMixin, Index):
     _typ = 'intervalindex'
     _comparables = ['name']
     _attributes = ['name', 'closed']
+    _allow_index_ops = True
+    _exception_rewrite = lambda: rewrite_exception('IntervalArray',
```

Comment:
Not sure what people's thoughts are. Without this, the exception would be

```python
In [6]: pd.IntervalIndex(2)
```

```pytb
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
TypeError: IntervalArray(...) must be called with a collection of some kind, 2 was passed
```

This lets the message be `IntervalIndex(...)`.